### PR TITLE
fix: missing define_variable for export declarations

### PR DESCRIPTION
**Pull request by xc2**
## Summary

fixes: #5994

## Require Documentation?

- [x] No
- [ ] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

**Review by h-a-n-a — APPROVED** [review]

**Comment by h-a-n-a** [comment]
Thank you!

File: crates/rspack_plugin_javascript/src/visitors/dependency/parser/walk_block_pre.rs [hunk 1]
```diff
@@ -30,7 +30,7 @@ impl<'parser> JavascriptParser<'parser> {
           }
           ModuleDecl::ExportNamed(decl) => self.block_pre_walk_export_name_declaration(decl),
           ModuleDecl::ExportDefaultExpr(_) => (),
```
Comment (ahabhgk) [inline crates/rspack_plugin_javascript/src/visitors/dependency/parser/walk_block_pre.rs:32]: seems `export default function Mod() {}` will have the same problem?

Comment (xc2) [reply crates/rspack_plugin_javascript/src/visitors/dependency/parser/walk_block_pre.rs:32]: > seems `export default function Mod() {}` will have the same problem?

`ExportDefaultDeclaration` case is covered here:

https://github.com/web-infra-dev/rspack/blob/ffd46f01b5985f8ae16dd36ab4faffc4a262741c/crates/rspack_plugin_javascript/src/visitors/dependency/parser/walk_block_pre.rs?plain=1#L28-L30

https://github.com/web-infra-dev/rspack/blob/ffd46f01b5985f8ae16dd36ab4faffc4a262741c/crates/rspack_plugin_javascript/src/visitors/dependency/parser/walk_block_pre.rs?plain=1#L93-L123

Comment (xc2) [reply crates/rspack_plugin_javascript/src/visitors/dependency/parser/walk_block_pre.rs:32]: test case added: https://github.com/web-infra-dev/rspack/pull/6024

**Pull request merged**